docs/features/asc1/stateful/hello_world.md

```diff
@@ -11,7 +11,7 @@ This is a very basic application which implements a counter. Each time the appli
 
 The [Overview](./index.md) of Algorand Stateful Smart Contracts should be consulted for details of the concepts covered in this document. Here is a brief primer to get started.
 
-The term "stateful" means the application is able to store information or "maintain state" within the ledger. The information ("data") is structured into _key/value pairs_. The [Transaction Execution Approval Language (TEAL)](../teal/index.md) defines the available [OpCodes](../../../reference/teal/opcodes.md) for use during program execution. _Application Call Transactions_ are used to interact with the application and may include _arguments_ (additional data) which are evaluated by the program at run-time. Every program execution must complete with a single non-zero unit64 value remaining on the stack to be valid and thus commit all state changes to the ledger. 
+The term "stateful" means the application is able to store information or "maintain state" within the ledger. The information ("data") is structured into _key/value pairs_. The [Transaction Execution Approval Language (TEAL)](../teal/index.md) defines the available [OpCodes](../../../reference/teal/opcodes.md) for use during program execution. _Application Call Transactions_ are used to interact with the application and may include _arguments_ (additional data) that are evaluated by the program at run-time. Every program execution must complete with a single non-zero unit64 value (or use the `return` opcode with a non-zero uint64 on top of the stack) remaining on the stack to be valid and thus commit all state changes to the ledger. 
```

Comment:
Last sentence is confusing. Suggestion:
Every program execution must complete with a single non-zero unit64 value remaining on the stack to be valid and for all state changes to be committed to the ledger.

The phrase in parens seemed redundant unless you mean that there could be multiple non-zero unit64 values on the stack and using `return` would be valid too.